Server/service/notificationService.js

```
@@ -16,6 +17,77 @@ class NotificationService {
 		this.logger = logger;
 	}
 
+	async sendDiscordNotification(networkResponse, webhookUrl) {
+        const { monitor, status } = networkResponse;
+        const message = {
+            content: `Monitor ${monitor.name} is ${status ? "up" : "down"}. URL: ${monitor.url}`
+        };
+
+        try {
+            await axios.post(webhookUrl, message);
+            return true;
+        } catch (error) {
+            this.logger.error({
+                message: error.message,
+                service: this.SERVICE_NAME,
+                method: "sendDiscordNotification",
+                stack: error.stack,
+            });
+            return false;
+        }
+    }
+
+	async sendSlackNotification(networkResponse, webhookUrl) {
+        const { monitor, status } = networkResponse;
+        const message = {
+            text: `Monitor ${monitor.name} is ${status ? "up" : "down"}. URL: ${monitor.url}`
+        };
+
+        try {
+            await axios.post(webhookUrl, message);
+            return true;
+        } catch (error) {
+            this.logger.error({
+                message: error.message,
+                service: this.SERVICE_NAME,
+                method: "sendSlackNotification",
+                stack: error.stack,
+            });
+            return false;
+        }
+    }
+
+	async sendTelegramNotification(networkResponse, address) {
+		const { monitor, status } = networkResponse;
+		
+		const [botToken, chatId] = address.split('|').map(part => part?.trim());
+		
+		if (!botToken || !chatId) {
+			return false;
+		}
+		
+		const message = {
+			chat_id: chatId,
+			text: `Monitor ${monitor.name} is ${status ? "up" : "down"}. URL: ${monitor.url}`,
+		};
+		
+		const url = `https://api.telegram.org/bot${botToken}/sendMessage`;
+		
+		try {
+			await axios.post(url, message, {
+				headers: {
+					'Content-Type': 'application/json',
+				},
+			});
+			return true;
+		} catch (error) {
+			return false;
```

I assume this won't be silently swallowed int the final implementation?

---

Gone.